### PR TITLE
Add Google Drive support to chitralekha

### PR DESCRIPTION
**Pull request by prakharrathi25**

### CODE REVIEW

File: backend/video/views.py
Comment (GokulNC) [inline backend/video/views.py:268]: Heyy, apart from these couple of lines, rest of the logic is same, right?
So can we please make them common?

**Review by GokulNC — CHANGES_REQUESTED** [review]

File: backend/video/views.py
Comment (prakharrathi25) [reply backend/video/views.py]: I have refactored the `list_recent` endpoint.

Comment (GokulNC) [reply backend/video/views.py]: Thanks!

Comment (prakharrathi25) [reply backend/video/views.py]: Can I merge this PR then?

Comment (GokulNC) [reply backend/video/views.py]: Uhh, I think we can merge this once the corresponding frontend changes for endpoint renaming is done.
Can you please inform Pranav about these endpoint that needs to be renamed? I think it should be simple changes.
Once done, we will merge & deploy both the frontend and backend together.

After that, we can sync-up with Harsh on Monday to explain how we can implement this import audio feature on the frontend. Pranav will not be able to work on this as we discussed earlier, since he has mid-sem exams from next week and may not be able to dedicate much time.

Comment (prakharrathi25) [reply backend/video/views.py]: Right right, I have already informed @pranav2305 about the changes.

File: backend/video/views.py
Comment (GokulNC) [inline backend/video/views.py:229]: If we are fixing with `is_audio_only`, please modify here as well.

As well as in the `get_video()` endpoint.

Comment (prakharrathi25) [reply backend/video/views.py]: Updated

File: backend/video/views.py
Comment (GokulNC) [inline backend/video/views.py:93]: This works only if the URL is of the following structure:
```
https://drive.google.com/file/d/1qJzF0CFH9atD4z32EMqsA7--aCD5kTt9/view
```

It won't work even if it is like this:
```
https://drive.google.com/file/d/1qJzF0CFH9atD4z32EMqsA7--aCD5kTt9
```

---

There can also be other formats, like:
```
https://drive.google.com/uc?id=1qJzF0CFH9atD4z32EMqsA7--aCD5kTt9
```

Can you please handle these cases? Also, if possible, explore what other formats it could be in

File: backend/video/views.py
Comment (GokulNC) [inline backend/video/views.py:101]: Suggestion -- Can you please try this: https://stackoverflow.com/a/22116732/
I am guessing it might become an overhead if the audio is too big.

Can you please try using this 30min audio (30MB) to see how much time it takes:
https://drive.google.com/file/d/1XT_KnkN2zAbIHEfSc49VCnN8xQr9ZnHF

If it isn't too time-consuming, let us maybe do it this way.
Also, the above answer is for python2. We might have to modify a few things for py3.

File: backend/video/views.py
Comment (GokulNC) [reply backend/video/views.py:93]: Basically, a way to standardize and store the normalized URL so that we don't create different records for same video (but with different link formats)

File: backend/video/views.py
Comment (GokulNC) [reply backend/video/views.py:101]: If this is too much, we can just put a TODO here saying that we need to come-up with a proper solution for this.

**Review by GokulNC — CHANGES_REQUESTED** [review]

File: backend/video/views.py
Comment (prakharrathi25) [reply backend/video/views.py]: I have dealt with a few cases in the `utils.extract_google_drive_link_id` which can be reviewed.

File: backend/video/views.py
Comment (prakharrathi25) [reply backend/video/views.py:107]: The time taken is not very high. The 30 minute video duration was calculated in 5 seconds.

File: backend/video/utils.py
Comment (GokulNC) [inline backend/video/utils.py:6]: Actually, since the file_id is a random alphanumeric string, there are good chances that even that file_id could have a substring `"id"`. Probably we can we can search for `"id="` itself?

Comment (GokulNC) [reply backend/video/utils.py:6]: Nevermind, actually found a better way as I was going through yt-dlp's code:

```py
>>> from yt_dlp.extractor import get_info_extractor
>>> drive_info_extractor = get_info_extractor("GoogleDrive")()
>>> drive_info_extractor._match_id("https://drive.google.com/uc?id=1XT_KnkN2zAbIHEfSc49VCnN8xQr9ZnHF")
'1XT_KnkN2zAbIHEfSc49VCnN8xQr9ZnHF'
```

We can have the `drive_info_extractor` object globally somewhere instead of creating each time.

If the match_id method throws an exception, we can return an error saying URL not supported.

Sorry for suggesting too many changes! :)
Should have thought this through properly.

File: backend/video/views.py
Comment (GokulNC) [inline backend/video/views.py]: Please remove `print`

File: backend/video/views.py
Comment (GokulNC) [inline backend/video/views.py:9]: Can you please add this to `requirements.txt`

File: backend/video/views.py
Comment (GokulNC) [inline backend/video/views.py:1]: Please remove `tracemalloc`

File: backend/video/views.py
Comment (GokulNC) [inline backend/video/views.py:109]: Just curious how it works - Is it actually a `filename`? Or `file` ?
If it's `file`, can we please rename to `file` to make it clear

**Review by GokulNC — CHANGES_REQUESTED** [review]

File: backend/video/views.py
Comment (prakharrathi25) [reply backend/video/views.py:109]: It is actually the filename - https://docs.python.org/3/library/urllib.request.html#urllib.request.urlretrieve

File: backend/video/views.py
Comment (prakharrathi25) [reply backend/video/views.py:1]: Done.

File: backend/video/views.py
Comment (prakharrathi25) [reply backend/video/views.py:9]: Yes yes.

File: backend/video/views.py
Comment (GokulNC) [reply backend/video/views.py:109]: Ohh OK.. One more question then:

```py
title = urllib.request.urlopen(urllib.request.Request(url)).info().get_filename()
```

Does this also download the full file just to get the filename?
Or does it manage to internally just read it using the HTTP header without having to download the file content?

Comment (prakharrathi25) [reply backend/video/views.py:109]: This doesn't download the full file. It uses the headers and metadata to read the filename

**Review by GokulNC — APPROVED** [review]
Thanks Prakhar! Will merge into dev.

**Pull request merged**
